### PR TITLE
Implement Google Docs-style link popover behavior

### LINKED ISSUE #188: [Bug]: URL popover sometimes shows up twice and out of place
### Contact Details

jim@refactorcoach.com

### What happened?

1. The URL link popover should currently display right under the Link button on the Coaching Notes toolbar. Sometimes (I don't know the pattern yet) it displays twice, one where it should and the other in the upper left hand corner of the browser window. It should only display once.

2. I'd also like to make this bug be an improvement as well. When one clicks on an existing link in the notes, which should display the link popover, I want it to show up above/below that link in the notes instead of under the Link toolbar button. However, it should still show up under the Link toolbar button when creating a new link.

3. It's also difficult to navigate coaching notes that have links with the mouse or text cursor because the link popover displays when it shouldn't. Let's modify it such that if the cursor is 1 space before or after (and still touching/adjacent to the link), it does not trigger the popover to display. Only if the cursor is actually in the link text. This is how other rich text editors behave as well.

Duplicate link popover screenshot:

Image

### What is the version of refactor-platform-fe that you're seeing the problem on?

1.0.0-beta2

### Any related code or configuration details to produce this issue as well as relevant log output.

```shell

```

### PR DESCRIPTION
**Pull request by jhodapp**
## Description
Implements Google Docs-style link popover behavior for the coaching notes editor. The link popover now intelligently opens when the cursor enters or touches a link (via keyboard or mouse), stays open while navigating within the link, and automatically closes when exiting. The popover no longer steals focus from the editor, allowing users to continue typing while viewing link information.

#### GitHub Issue: Resolves #188

**Note:** This is based on another branch, [land this PR](https://github.com/refactor-group/refactor-platform-fe/pull/205) first before this one.

### Changes
* Added boundary detection logic (`isCursorInsideLink`) to determine if cursor is inside or touching link content
* Added left-boundary trigger: popover now opens when cursor is positioned just to the left of a link (touching it)
* Implemented URL extraction for links when cursor is touching from the left
* Implemented state tracking (`wasInsideLink` ref) to only trigger popover on entering a link, preventing aggressive reopening after user closes it
* Added auto-close behavior (`onLinkExit`) that closes popover when cursor navigates from inside link to outside
* Removed early `editor.isActive("link")` check to enable left-boundary detection
* Changed link cursor style from `pointer` to `text` in `simple-editor.scss` to maintain text editing context
* Added `initialFocus={-1}` to `PopoverContent` to prevent focus stealing from editor
* Extended `PopoverContent` component with `modal` and `initialFocus` props for better focus management
* Added `typecheck` npm script to package.json
* Updated component tests to reflect new Google Docs-style behavior (14 tests, all passing)

### Screenshots / Videos Showing UI Changes (if applicable)
Screenshots would be helpful showing:
1. Cursor navigating into a link via keyboard arrow keys - popover appears
2. Cursor positioned just to the left of link (touching) - popover appears with correct URL
3. Cursor at first/last character position in link - popover is visible
4. Cursor navigating from inside link to outside - popover auto-closes
5. Typing in editor while popover is open - cursor remains in editor, not stolen by popover
6. Mouse hovering over link - text cursor shown (not pointer hand)
7. Clicking outside popover - popover closes and stays closed

### Testing Strategy
**Component Tests**: Run `npm test -- __tests__/components/ui/tiptap-ui/link-popover.test.tsx`
- 14 tests covering single popover rendering, positioning, Google Docs-style auto-open behavior, state management, and edge cases

**Manual Testing**:
1. Navigate to any coaching session with notes editor
2. Create or click on an existing link
3. Use keyboard arrow keys to navigate into the link → verify popover opens
4. Move cursor within link → verify popover stays open
5. Move cursor out of link → verify popover auto-closes
6. Position cursor just to the left of link → verify popover opens with correct URL
7. Navigate back into link → verify popover opens again
8. While popover is open, try typing in the editor → verify cursor stays in editor
9. Click outside the popover → verify it closes and doesn't reopen
10. Hover mouse over link → verify text cursor (I-beam) is shown, not pointer hand

### Concerns
None - the implementation follows the established patterns and all existing tests continue to pass.

### CODE REVIEW

File: src/components/ui/tiptap-ui-primitive/popover/popover.tsx [hunk 1]
```diff
@@ -264,6 +264,8 @@ interface PopoverContentProps extends React.HTMLProps<HTMLDivElement> {
   portal?: boolean
   portalProps?: Omit<React.ComponentProps<typeof FloatingPortal>, "children">
   asChild?: boolean
+  modal?: boolean
+  initialFocus?: number | React.MutableRefObject<HTMLElement | null>
```
Comment (calebbourg) [inline src/components/ui/tiptap-ui-primitive/popover/popover.tsx]: I'm not sure I understand why `React.MutableRefObject<HTMLElement | null>` is used here or what it's for. If this is idiomatic maybe we can document why it's used? Otherwise perhaps we can use something more primitive?

**Review by calebbourg — CHANGES_REQUESTED** [review]
Just a few suggestions

File: src/components/ui/tiptap-ui-primitive/popover/popover.tsx (hunk 1 above)
Comment (jhodapp) [reply src/components/ui/tiptap-ui-primitive/popover/popover.tsx]: Great suggestion.
